### PR TITLE
Only ship repo files for the distro being built

### PR DESCRIPTION
**Pull request by bcl**
This is my modification of pr #1705 

This pull request includes:

- [x] adequate testing for the new functionality or fixed issue

### CODE REVIEW

File: osbuild-composer.spec [hunk 1]
```diff
@@ -155,8 +155,21 @@ install -m 0755 -vp _bin/osbuild-composer                       %{buildroot}%{_l
...
+# Only include repositories for the distribution and release
 install -m 0755 -vd                                             %{buildroot}%{_datadir}/osbuild-composer/repositories
-install -m 0644 -vp repositories/*                              %{buildroot}%{_datadir}/osbuild-composer/repositories/
+# CentOS also defines rhel so we check for centos first
+%if 0%{?centos}
+install -m 0644 -vp repositories/centos-%{centos}*              %{buildroot}%{_datadir}/osbuild-composer/repositories/
+install -m 0644 -vp repositories/centos-stream-%{centos}*       %{buildroot}%{_datadir}/osbuild-composer/repositories/
+%else
+%if 0%{?rhel}
+install -m 0644 -vp repositories/rhel-%{rhel}*                  %{buildroot}%{_datadir}/osbuild-composer/repositories/
```
Comment (teg) [inline osbuild-composer.spec:166]: We probably do want to allow building RHEL8 on a RHEL9 host. I.e., backwards compatibility should be fine, forwards compatibility would be harder (i.e., doesn't work, so better not expose it).

**Review by teg — CHANGES_REQUESTED** [review]
Thanks a lot for this!

**Comment by bcl** [comment]
I don't know what's up with schutzbot, but it should be run before this is merged.

**Review by teg — DISMISSED** [review]
Nice! Thanks!

File: test/cases/cross-distro.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+fi
+
+if ! DISTROS=$(curl -s --unix-socket $APISOCKET http:///localhost/api/v1/distros/list); then
+    echo "osbuild-composer server error getting distros list. 😢"
+    exit 1
+fi
+
+REMAINDER=$(echo "$DISTROS" | jq -r '.distros' | grep -v -E "$PATTERN")
+if [ -n "$REMAINDER" ]; then
+	echo "🔥 Unexpected distros installed:"
```
Comment (ochosi) [inline test/cases/cross-distro.sh:70]: Indentation error?

Comment (bcl) [reply test/cases/cross-distro.sh]: yeah, some tabs snuck back in. should be fixed now.

**Pull request merged**